### PR TITLE
fix(typescript-estree): correct parenthesized optional chain AST

### LINKED ISSUE #1139: Optional chaining: (a?.b).c is parsed as a?.b.c
https://github.com/prettier/prettier/pull/6657 is blocked by this issue.

**What code were you trying to parse?**

```ts
(a?.b).c
```

**What did you expect to happen?**

The outer node should be of type `MemberExpresion`, not `OptionalMemberExpression`.

**What actually happened?**

Both member expressions are `OptionalMemberExpression` as if the parens weren't there (`a?.b.c`).

**Versions**

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/typescript-estree` | `2.5.1-alpha.3` |
| `TypeScript`                           | `3.7.0-rc` |
| `node`                                 | `10.15.0` |
| `npm`                                  | `6.11.3` |

### PR DESCRIPTION
**Pull request by bradzacher**
Fixes #1139

Also fixes the package not working in the browser (`ts.sys === undefined` in the browser).

The snapshots are getting pretty ugly now.
I think we need a new system for them (#1142).

I tested this via a local build of ASTExplorer.
Sidebar - we should really maintain our own version of it (#1143).

### CODE REVIEW

**Comment by thorn0** [comment]
Please update also `SUPPORTED_TYPESCRIPT_VERSIONS`. It doesn't include [`3.1.0-rc`](https://www.npmjs.com/package/typescript/v/3.7.1-rc).

**Comment by thorn0** [comment]
`warnAboutTSVersion` also fails in the browser because of `process.stdout.isTTY`.

File: packages/typescript-estree/src/parser.ts [hunk 1]
```diff
@@ -224,22 +231,21 @@ function applyParserOptionsToExtra(options: TSESTreeOptions): void {
...
-      'WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.',
-      'You may find that it works just fine, or you may not.',
-      `SUPPORTED TYPESCRIPT VERSIONS: ${SUPPORTED_TYPESCRIPT_VERSIONS}`,
-      `YOUR TYPESCRIPT VERSION: ${ACTIVE_TYPESCRIPT_VERSION}`,
-      'Please only submit bug reports when using the officially supported version.',
-      border,
-    ];
-    extra.log(versionWarning.join('\n\n'));
+  if (!isRunningSupportedTypeScriptVersion && !warnedAboutTSVersion) {
+    const isTTY = typeof process === undefined ? false : process.stdout.isTTY;
```
Comment (thorn0) [inline packages/typescript-estree/src/parser.ts]: In Prettier's build configuration, we have `process`, but not `process.stdout`.  🤷‍♂️

Comment (bradzacher) [reply packages/typescript-estree/src/parser.ts]: darn...

probably just better to do what `parser` does, and clear the logger function:

https://github.com/typescript-eslint/typescript-eslint/blob/d4703e1a97760604cadb4167d5de4a897883119f/packages/parser/src/parser.ts#L87-L93

(note that this logger function isn't used anywhere else, it's only used to log the warning)

**Pull request merged**